### PR TITLE
[DF] Prefer US spelling for initialize/finalize

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

**Comment by eguiraud** [comment]
Rebased to fix a minor conflict.

File: tree/dataframe/inc/ROOT/RDataSource.hxx [hunk 1]
```diff
@@ -190,31 +191,46 @@ public:
...
    /// users can run multiple event-loops with the same RDataFrame.
-   /// Ideally, `Initialise` should set the state of the RDataSource so that multiple identical event-loops
+   /// Ideally, `Initialize` should set the state of the RDataSource so that multiple identical event-loops
    /// will produce identical results.
    // clang-format on
-   virtual void Initialise() {}
+   virtual void Initialize() {}
+
+   /// \cond
+   R__DEPRECATED(6,30,"Use Initialize (with a z) instead.") void Initialise() { Initialize(); }
```
Comment (Axel-Naumann) [inline tree/dataframe/inc/ROOT/RDataSource.hxx:200]: Doesn't this need to be `virtual` to not break expert users' derived classes? And even then... someone needs to call it?

File: tree/dataframe/src/RLoopManager.cxx [hunk 2]
```diff
@@ -504,7 +504,7 @@ void RLoopManager::RunTreeReader()
 void RLoopManager::RunDataSource()
 {
    assert(fDataSource != nullptr);
-   fDataSource->Initialise();
```
Comment (Axel-Naumann) [inline tree/dataframe/src/RLoopManager.cxx:507]: You probably need to call both versions here, for backward compatibility? You can even test that by moving only half the data sources to the new spelling and checking whether both new and old still work.

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

**Comment by eguiraud** [comment]
As per discussion with Axel:

- [ ] instead of `R__DEPRECATE`, use a runtime printout: it's not possible to have `R__DEPRECATE` fire for users that inherit from `RDataSource` but not for ROOT builds, and maintain backwards-compatibility. But it's possible with a runtime error.
- [ ] put a R__DEPRECATE in there even if it does not fire, e.g. as an attribute of an unused dummy struct.

File: tree/dataframe/inc/ROOT/RDataSource.hxx [hunk 3]
```diff
@@ -190,31 +192,86 @@ public:
...
+   };
+
+   virtual void Initialise() { throw 0; }
+
+   void CallInitialize()
+   {
+      try {
+         Initialise();
+      } catch (int) {
+         Initialize();
```
Comment (Axel-Naumann) [inline tree/dataframe/inc/ROOT/RDataSource.hxx:215]:
```suggestion
         // `Initialise()` was not overridden and threw, the data source uses the new spelling: good!
         Initialize();
```

File: tree/dataframe/inc/ROOT/RDataSource.hxx [hunk 4]
```diff
@@ -190,31 +192,86 @@ public:
...
+         return;
+      }
+      Warning("RDataSource::Initialise",
+              "Initialise is deprecated. Please implement Initialize (with a z) instead of Initialise.");
```
Comment (Axel-Naumann) [inline tree/dataframe/inc/ROOT/RDataSource.hxx:219]:
```suggestion
              "Initialise is deprecated. Please rename it to \"Initialize\" (with a z).");
```
?

**Review by Axel-Naumann — APPROVED** [review]
LGTM! Can you already prepare a PR to merge at the time of v6.30, i.e. removing the deprecation etc, and can you maybe reference that from the comment near `NeverUsedJustAReminder`?

**Pull request merged**